repost/api/resolvers.py

```diff
@@ -94,28 +94,39 @@ def resolve_user(username: str = Path(...), db: Session = Depends(get_db)) -> mo
 
     Base path: /resubs/{resub}/posts/{post_id}
     """
-    if (resub.owner_id != current_user.id) and post.author_id != current_user.id:
+    if (resub.owner_id != current_user.id) and (post.author_id != current_user.id):
```

Comment:
```suggestion
    if current_user not in (post.author, resub.owner):
```
Comparisons can be done on the relationship models directly instead of on ID, and this pattern might be more readable.